MCP-OCP-Lightspeed (#894)
kubev2v/forklift-documentation: documentation/modules/con_using-lightspeed-with-mtv.adoc

Real-time troubleshooting:: Get plain-language explanations of migration failures, error messages, and stalled migrations without manually parsing logs.
+
For example:
+
* "Why did the migration for VM 'ProdDB-01' fail?"
* "Why is my warm migration stalled?"
* "What does the error 'import retry limit exceeded' mean?"
* "How do I resolve Changed Block Tracking (CBT) snapshot issues?"

Post-migration guidance:: Receive recommendations for validating migrated virtual machines, optimizing resource configurations, and troubleshooting post-migration issues.
+
For example:
+
* "Why aren't my VMs working after migration?"
* "How do I verify my migrated VMs are running correctly?"
* "What should I check after migration is complete?"

== Data access and analysis

When you ask {lightspeed-short} a question about {project-short}, {lightspeed-short}:

. Accesses real-time data from your cluster, including migration plan status, custom resources, and logs
. Analyzes the data in accordance with your user credentials and role-based access control (RBAC) permissions
. Provides a conversational response with specific, actionable information

== Privacy and data handling

> {lightspeed-short} operates by using your own user credentials and can only access resources that you have permission to view. {lightspeed-short} accesses cluster metadata, migration resource status, and log data to provide contextual assistance. All data access respects your existing RBAC permissions, and {lightspeed-short} cannot access resources beyond your authorization level.

Reviewer: that is true, lightspeed can't access information you can't
but it can access information you have access to, so depending on your setup, you might expose your information to the unauthorized people who provider the lightspeed service.

do we want to tell users that using AI can be inherently insecure if the AI provider ( google, ms, anthropic ) is not contractually obligated to  keep your information ?
a w/a is use "in house" AI providers like on premiss RHEL-AI or Openshift-AI where all the information is kept on premiss. 

do we need to add a note about potential AI data leakage when using external AI service ?

Reviewer: @mnecas ^^ WDYT ?

Author: Added note

== How the integration works

> After you enable the `feature_mcp_server` feature flag in the `ForkliftController` custom resource, the {project-short} Operator configures the integration with {lightspeed-full} by:

Author: `feature_mcp_server` is the correct flag, as referenced in:
  - con_using-lightspeed-with-mtv.adoc:63 - In the concept explanation
  - proc_enabling-lightspeed-integration.adoc:33 - In the procedure step
  - proc_enabling-lightspeed-integration.adoc:38 - In the YAML example

// Module included in the following assemblies:
//
// * documentation/doc-Planning_your_migration/assemblies/assembly_lightspeed-integration.adoc

:_mod-docs-content-type: CONCEPT

[id="con_using-lightspeed-with-mtv_{context}"]

> = {lightspeed-full} integration with {project-short}

Reviewer 2: Follow on from - https://github.com/kubev2v/forklift-documentation/pull/894/files#r3402692105

Duplicate titles — assembly and concept module read identically assembly_lightspeed-integration.adoc and con_using-lightspeed-with-mtv.adoc
Both have = {lightspeed-full} integration with {project-short}. When the concept is included with leveloffset=+1, the reader sees the same heading text at two consecutive heading levels. Give the concept a distinct title, for example:

= About {lightspeed-short} and {project-short}

[role="_abstract"]
Use {lightspeed-full} to quickly diagnose migration issues and get step-by-step guidance without manually parsing logs or searching documentation.

> {lightspeed-full} is integrated into the {ocp} web console. It provides intelligent assistance through a natural-language interface, allowing you to ask questions, receive troubleshooting guidance, and get step-by-step instructions.

Reviewer 2: ```suggestion
{lightspeed-full} is integrated into the {ocp} web console. It provides intelligent assistance through a natural-language interface, enabling you to ask questions, receive troubleshooting guidance, and get step-by-step instructions.
```

Author: It provides intelligent assistance through a natural-language interface where you can ask questions, receive troubleshooting guidance, and get step-by-step instructions.

+
For example:
+
* "What are the prerequisites for warm migration from VMware?"
* "What network requirements do I need for migrating VMs?"
* "How do I check network mapping for a failed migration?"

Real-time troubleshooting:: Get plain-language explanations of migration failures, error messages, and stalled migrations without manually parsing logs.
+
For example:
+
* "Why did the migration for VM 'ProdDB-01' fail?"
* "Why is my warm migration stalled?"
* "What does the error 'import retry limit exceeded' mean?"
* "How do I resolve Changed Block Tracking (CBT) snapshot issues?"

Post-migration guidance:: Receive recommendations for validating migrated virtual machines, optimizing resource configurations, and troubleshooting post-migration issues.
+
For example:
+
* "Why aren't my VMs working after migration?"
* "How do I verify my migrated VMs are running correctly?"
* "What should I check after migration is complete?"

== Data access and analysis

When you ask {lightspeed-short} a question about {project-short}, {lightspeed-short}:

. Accesses real-time data from your cluster, including migration plan status, custom resources, and logs

> . Analyzes the data in accordance with your user credentials and role-based access control (RBAC) permissions

Reviewer 2: ```suggestion
. Analyzes the data based on your user credentials and role-based access control (RBAC) permissions
```

. Provides a conversational response with specific, actionable information

== Privacy and data handling

> {lightspeed-short} operates by using your own user credentials and can only access resources that you have permission to view. {lightspeed-short} accesses cluster metadata, migration resource status, and log data to provide contextual assistance. All data access respects your existing RBAC permissions, and {lightspeed-short} cannot access resources beyond your authorization level.

Reviewer 2: ```suggestion
{lightspeed-short} uses your own user credentials and can only access resources that you have permission to view. {lightspeed-short} accesses cluster metadata, migration resource status, and log data to provide contextual assistance. All data access respects your existing RBAC permissions, and {lightspeed-short} cannot access resources beyond your authorization level.
```

[IMPORTANT]
====

> When using Lightspeed with an external AI service provider, cluster information, logs, and resource details may be transmitted to that provider. Evaluate whether external AI services align with the security and compliance requirements of your organization.

Reviewer 2: ```suggestion
When you use {lightspeed-short} with an external AI service provider, cluster information, logs, and resource details may be transmitted to that provider. Evaluate whether external AI services align with the security and compliance requirements of your organization.
```
